File: backend/internal/domain/order.go
package domain

import "time"

type OrderStatus string

const (
	OrderStatusPending    OrderStatus = "pending"
	OrderStatusConfirmed  OrderStatus = "confirmed"
	OrderStatusPreparing  OrderStatus = "preparing"
	OrderStatusReady      OrderStatus = "ready"
	OrderStatusDelivered  OrderStatus = "delivered"
	OrderStatusCancelled  OrderStatus = "cancelled"
)

type Order struct {
	ID         uint
	Status     OrderStatus
	TotalPrice float64
	Notes      string
	CreatedAt  time.Time
	UpdatedAt  time.Time

	// Preenchido sob demanda
	Items []OrderItem
}

File: frontend/src/lib/types/order.ts
export type OrderStatus = 'pending' | 'confirmed' | 'preparing' | 'ready' | 'delivered' | 'cancelled';

export interface OrderItem {
  ID: number;
  OrderID: number;
  ProductID: number;
  Quantity: number;
  UnitPrice: number;
  Product?: any; // pode ser null ou objeto completo
}

export interface Order {
  ID: number;
  Status: OrderStatus;
  TotalPrice: number;
  Notes?: string;
  Items: OrderItem[];
  CreatedAt?: string;
  UpdatedAt?: string;
}

export type OrderCreatePayload = {
  notes?: string;
  items: { product_id: number; quantity: number }[];
};

export const ORDER_STATUS_LABEL: Record<OrderStatus, string> = {
  pending:    'Pendente',
  confirmed:  'Confirmado',
  preparing:  'Preparando',
  ready:      'Pronto',
  delivered:  'Entregue',
  cancelled:  'Cancelado',
};

export const ORDER_STATUS_COLOR: Record<OrderStatus, string> = {
  pending:   'badge-warning',
  confirmed: 'badge-info',
  preparing: 'badge-info',
  ready:     'badge-success',
  delivered: 'badge-neutral',
  cancelled: 'badge-error',
};
